core/Ajax.php

```diff
@@ -453,26 +450,42 @@ public static function view_client_list($clientId = '')
                 $html .= 'Create new list';
                 $html .= '</option>';
                 $html .= '<option class="ajax-call" disabled >';
-                $html .= '---';
+                $html .= '--- Single opt-in --- suitable for this plugin';
                 $html .= '</option>';
                 $isSelected = false;
+                $hasConfirmedOptInList = false;
+                $htmlConfirmedOptIn = '';
                 foreach ($lists as $list) {
                     $id = $list->ListID;
-
+                    
                     $viewClientListUrl = http_build_query((array)$list);
-                    $fields = App::$CampaignMonitor->get_stats($id);
+                    $listDetails = App::$CampaignMonitor->get_list_details($id);
                     $selected = '';
 
-                    if ($id == $selectedList){
-                        $selected = 'selected="selected"';
-                        $isSelected = true;
+                    if (!$listDetails->ConfirmedOptIn) {
+                        if ($id == $selectedList){
+                            $selected = 'selected="selected"';
+                            $isSelected = true;
+                        }
+   
+                        $html .= '<option '.$selected .' value="'.$clientId.'" data-id="'.$id.'"  data-url="' . self::$actionUrl . '&' . $viewClientListUrl . '&ClientID=' . $clientId . '&action=get_list_settings">';
+                        $html .= Util::htmlDecodeEncode($list->Name);
+                        $html .= '</option>';
                     }
+                    else {
+                        $hasConfirmedOptInList = true;
+                        $htmlConfirmedOptIn .= '<option '.$selected .' disabled value="'.$clientId.'" data-id="'.$id.'"  data-url="' . self::$actionUrl . '&' . $viewClientListUrl . '&ClientID=' . $clientId . '&action=get_list_settings">';
+                        $htmlConfirmedOptIn .= Util::htmlDecodeEncode($list->Name);
+                        $htmlConfirmedOptIn .= '</option>';
+                    }
+                }
 
-
-                    $html .= '<option '.$selected .' value="'.$clientId.'" data-id="'.$id.'"  data-url="' . self::$actionUrl . '&' . $viewClientListUrl . '&ClientID=' . $clientId . '&action=get_list_settings">';
-                    $html .= Util::htmlDecodeEncode($list->Name);
+                if ($hasConfirmedOptInList) {
+                    $html .= '<option class="ajax-call" disabled >';
+                    $html .= '--- Confirmed opt-in --- not suitable for this plugin';

```

Comment:
Thank you for adding the front end validation in the list, but I noticed that you can still save a double pot in list by re-enable the option in the list.
Optional: Maybe good to add backend validation as well to prevent user from saving invalid option?

Also noticed when you have double opt in list selected previously, the list will no longer show the current selected list? Is it intentional? 
![image](https://user-images.githubusercontent.com/33608331/200224368-3b29629f-1e36-42cf-b9ed-068b76ca8678.png)

Reply:
It was intentional... in my initial attempt, I had let it that you could still keep the old setting. However, given the severity of the issue - I then make it that you can't keep the previously double opt-in setting (and thus adding the front-end validation too as otherwise it throws a silent error otherwise). 

I had briefly thought about it and since I'm intending to make minimal changes to it, I had left it as is... In order to add backend validation - we would need to either update the actual API implementation (which isn't ideal as it's major change) OR alternatively making an extra call to re-verify the list setting. Further to that, it would felt a little bit complex in having the front-end handling this specfic error (should we put a red-box again, but this is the result of hacking too). Given this is already edge-case, I left it with only front-end validation. 

Let me know your thought if you still think that it's still worthwhile to update the backend - we can have a chat about it.

Reply:
Sure, it might be nice to a extra front end validation here so even user re-enable the option, they can not submit. But I will leave it you if it is too much work.

For the second point, I think we should still consider to show the selected list name in the dropdown but give a front end validation or a warning/error message to ask the user to reset their settings.
This will help user to understand why their list is no longer shown here and promote them to change their list.
They should all be front end changes so should not be to much work I hope, what do you think?

Reply:
![image](https://user-images.githubusercontent.com/41770203/200475768-bd35c1c6-e586-4775-b1b2-3373e83e807a.png)
Adding a warning/message about the previously confirmed-opt-in message